web/src/components/Chat.tsx

```diff
@@ -107,7 +106,11 @@ const Chat = ({ messages, submitMessage, hasError, isLoading }: ChatProps): Reac
           />
         </InputSection>
         <SubmitContainer>
-          <SubmitButton disabled={textInput.length === 0} onClick={onSubmit} text={t('sendButton')} />
+          <SubmitButton
+            disabled={textInput.length === 0 || hasError || isLoading}
```

Comment:
🔧 This logic is duplicated, please extract to a constant:
```ts
  const submitDisabled = textInput.trim().length === 0 || hasError || isLoading
  const submitOnEnter = (event: KeyboardEvent) => {
    const isEnterAllowed = event.key === 'Enter' && !event.shiftKey && !submitDisabled
    ...
```
```suggestion
            disabled={submitDisabled}
```